preview/MsixCore/msixmgr/Package.cpp

```diff
@@ -148,30 +155,120 @@ HRESULT MsixCoreLib::PackageBase::ProcessPSFIfNecessary()
+                    psfExecutable.assign(executable.begin(), executable.end());
+
+                    std::string arguments = (*itr)["arguments"].GetString();
+                    std::wstring psfArguments;
+                    psfArguments.assign(arguments.begin(), arguments.end());
+
+                    std::string workingDirectory = (*itr)["workingDirectory"].GetString();
+                    std::wstring psfWorkingDirectory;
+                    psfWorkingDirectory.assign(workingDirectory.begin(), workingDirectory.end());
+
+                    m_executionInfo.commandLineArguments = psfArguments;
+
+                    // resolve the given paths from json into full paths.
+                    m_executionInfo.resolvedExecutableFilePath = FilePathMappings::GetInstance().GetExecutablePath(psfExecutable, m_packageFullName.c_str());
+                    m_executionInfo.workingDirectory = FilePathMappings::GetInstance().GetExecutablePath(psfWorkingDirectory, m_packageFullName.c_str());
+
+                    if ((*itr).HasMember("startScript"))
+                    {
+                        const Value& startScript = (*itr)["startScript"];
+
+                        std::string scriptPath = startScript["scriptPath"].GetString();
+                        std::wstring psfScriptPath;
+                        psfScriptPath.assign(scriptPath.begin(), scriptPath.end());
+                        m_scriptSettings.scriptPath = psfScriptPath;
+
+                        if (startScript.HasMember("runOnce"))
+                        {
+                            bool runOnce = startScript["runOnce"].GetBool();
+                            m_scriptSettings.runOnce = runOnce;
+
+                            if (runOnce == false)
+                            {
+                                return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
+                            }
+                        }
+                        else
+                        {
+                            m_scriptSettings.runOnce = false;
+                        }
+
+                        if (startScript.HasMember("showWindow"))
+                        {
+                            bool showWindow = startScript["showWindow"].GetBool();
+                            m_scriptSettings.showWindow = showWindow;
+                        }
+                        else
+                        {
+                            m_scriptSettings.showWindow = false;
+                        }
+
+                        if (startScript.HasMember("waitForScriptToFinish"))
+                        {
+                            bool waitForScriptToFinish = startScript["waitForScriptToFinish"].GetBool();
+                            m_scriptSettings.waitForScriptToFinish = waitForScriptToFinish;
+                        }
+                        else
+                        {
+                            m_scriptSettings.waitForScriptToFinish = false;

```

Comment:
>false [](start = 69, length = 5)

this also defaults to true (see the table [https://github.com/microsoft/MSIX-PackageSupportFramework](https://github.com/microsoft/MSIX-PackageSupportFramework))

Reply:
Removed else case since its initialized now. If name-value pair is absent, it will default to respective value.